homeassistant/data_entry_flow.py

```diff
@@ -829,6 +830,14 @@ def async_show_progress(
             flow_result["step_id"] = step_id
         return flow_result
 
+    @callback
+    def async_update_progress(self, progress: float) -> None:
+        """Update the progress of a flow."""
```

Comment:
We should document here what the `progress` parameter means and what the minimum and maximum progress is.